source/fundamentals/data-formats/polymorphism.txt

```diff
@@ -0,0 +1,271 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To simplify queries against your collection of polymorphic types, you can use the
+``HierarchicalDiscriminatorConvention``. According to this convention, the value of ``_t``
+is an array of all classes in the inheritance hierarchy of the document's type.
+
+To use the ``HierarchicalDiscriminatorConvention``, label the base class of your
+inheritance hierarchy as the root class. If you're using the automapper,
+label the root class by applying the
+``[BsonDiscriminatorAttribute]`` attribute to the class and passing ``RootClass = true``
+as an argument. The following code example labels the ``Animal`` class as the
+root of the example inheritance hierarchy:
+
+.. code-block:: csharp
+   :emphasize-lines: 1
+   
+   [BsonDiscriminator(RootClass = true)]
+   [BsonKnownTypes(typeof(Cat), typeof(Dog)]
+   public class Animal 
+   {
+   }
+
+If you're creating a class map manually, call the ``SetIsRootClass()`` method and pass
+``true`` as an argument when you register the class map for the root class. The following
+code example registers class maps for all five example classes but labels only the
+``Animal`` class as the root of the inheritance hierarchy:
+
+.. code-block:: csharp
+   :copyable: true
+   :emphasize-lines: 3 
+   
+   BsonClassMap.RegisterClassMap<Animal>(classMap => {
+       classMap.AutoMap();
+       classMap.SetIsRootClass(true);
+   });
+   BsonClassMap.RegisterClassMap<Cat>();
+   BsonClassMap.RegisterClassMap<Dog>();
+   BsonClassMap.RegisterClassMap<Lion>();
+   BsonClassMap.RegisterClassMap<Tiger>();
+
+Suppose you labeled the example ``Animal`` class as the root of the inheritance hierarchy,
+and then created an instance of the ``Animal`` class and each of its
+subclasses. If you serialized these objects to a single collection, the {+driver-short+}
+would apply the ``HierarchicalDiscriminatorConvention`` and the corresponding
+BSON documents would appear as follows:
+
+.. code-block:: javascript
+   
+   { _id: ..., _t: "Animal", ... }
+   { _id: ..., _t: ["Animal", "Cat"], ... }
+   { _id: ..., _t: ["Animal", "Dog"], ... }
+   { _id: ..., _t: ["Animal", "Cat", "Lion"], ... }
+   { _id: ..., _t: ["Animal", "Cat", "Tiger"], ... }
+
+.. important:: Root Class Discriminator
+
+   Any document mapped to the root class still uses a string value for the discriminator
+   field. 
+
+When using the ``HierarchicalDiscriminatorConvention``, you can search for all
+documents of type or subtype ``Cat`` by using a single boolean condition, as shown in
+the following example:
+
+.. code-block:: csharp
+   :copyable: true
+   
+   var query = coll.Aggregate().Match(a => a is Cat);
+
+Custom Discriminator Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you're working with data that doesn't follow the conventions used by the
+{+driver-short+}--for example, data inserted into MongoDB by another driver or object
+mapper--you might need to use a different value for your discriminator field to
```

Comment:
I don't know how to rewrite this one - I prefer "might" to "may" and I don't actually see where in the style guide this is written as a rule.

Reply:
yeah, "must" doesn't have a direct subjunctive equivalent. leaving as is